File: frontend/src/types/order.ts
// src/types/order.ts
export interface OrderItem {
    id: number;
    product_id: number | null;
    product_name: string | null;
    quantity: number;
    price: number;
    product?: {
      id: number;
      name: string;
      image_url: string | null;
    } | null;
  }
  
  export interface Order {
    id: number;
    order_number: string;
    total_amount: number;
    status: string;
    shipping_address: string;
    created_at: string;
    updated_at: string;
    items: OrderItem[];
  }
  
  export interface OrderCreate {
    shipping_address: string;
  }
  
  export interface OrderListResponse {
    items: Order[];
    total: number;
  }

  export interface AdminOrder extends Order {
    user?: {
      id: number;
      username: string;
      email: string;
    };
  }
  
  export interface AdminOrderListResponse {
    items: AdminOrder[];
    total: number;
  }

File: backend/app/schemas/order.py
# app/schemas/order.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional
from app.schemas.product import ProductOut

# ========== 订单项 ==========
class OrderItemOut(BaseModel):
    id: int
    product_id: Optional[int] = None
    product_name: Optional[str] = None
    quantity: int
    price: float
    product: Optional[ProductOut] = None

    class Config:
        from_attributes = True

class UserBrief(BaseModel):
    id: int
    username: str
    email: str

    class Config:
        from_attributes = True

# ========== 订单 ==========
class OrderOut(BaseModel):
    id: int
    order_number: str
    total_amount: float
    status: str
    shipping_address: str
    created_at: datetime
    updated_at: datetime
    items: list[OrderItemOut] = []
    user: Optional[UserBrief] = None

    class Config:
        from_attributes = True

class OrderListResponse(BaseModel):
    items: list[OrderOut]
    total: int

class OrderCreate(BaseModel):
    shipping_address: str = Field(..., min_length=5, max_length=500, description="收货地址")
    coupon_code: Optional[str] = Field(None, description="优惠券码（可选）")

class OrderDiscountInfo(BaseModel):
    coupon_code: str
    coupon_name: str
    discount_amount: float

class OrderStatusUpdate(BaseModel):
    status: str = Field(..., description="新状态: pending/paid/shipped/completed/cancelled")
